pkgs/development/tools/parsing/bison/default.nix

```diff
@@ -18,6 +18,11 @@ stdenv.mkDerivation rec {
   # Otherwise tests fail due to incorrect unicode symbol oconversion.
   configurePlatforms = [ "build" "host" ];
 
+  # there's a /bin/sh shebang in bin/yacc which when no strictDeps is patched with the build stdenv shell
+  # however when cross-compiling it would still be patched with the build stdenv shell which would be wrong
+  # cannot add bash to buildInputs due to infinite recursion
+  strictDeps = stdenv.hostPlatform != stdenv.buildPlatform;
```

Comment:
I would do
```nix
buildInputs = [
  # Hack, cannot depend on bash in the native build without
  # causing infinite recursion during bootstrapping
  (if stdenv.hostPlatform != stdenv.buildPlatform then bash else stdenv.shell)
];
```
instead

Reply:
```
$ nix build ".#bison"
bison> /nix/store/nmca9y3m39hypn342bwzwm6c4yd5y9lp-bootstrap-stage1-stdenv-linux/setup: line 528: source: /nix/store/p4s4jf7aq6v6z9iazll1aiqwb34aqxq9-bootstrap-tools/bin/bash: cannot execute binary file
note: keeping build directory '/comp-temp/nix-build-bison-3.8.2.drv-4'
error: builder for '/nix/store/xgmr7z154q3fmbkyr97m0na9jcyjvixf-bison-3.8.2.drv' failed with exit code 126;
       last 1 log lines:
       > /nix/store/nmca9y3m39hypn342bwzwm6c4yd5y9lp-bootstrap-stage1-stdenv-linux/setup: line 528: source: /nix/store/p4s4jf7aq6v6z9iazll1aiqwb34aqxq9-bootstrap-tools/bin/bash: cannot execute binary file
       For full logs, run 'nix log /nix/store/xgmr7z154q3fmbkyr97m0na9jcyjvixf-bison-3.8.2.drv'.
```

https://github.com/NixOS/nixpkgs/blob/7283d3942f7947949a131cfa81e5fd0688af3de3/pkgs/stdenv/generic/setup.sh#L515

Reply:
```
nix-repl> stdenv.shell
"/nix/store/wpcyy8zajlr47y0vmljwj3ppgck0rb7k-bash-5.1-p16/bin/bash"

nix-repl> stdenv.shellPackage
«derivation /nix/store/5qnnrqd025j0swnvhiqxg7sda2x9n9yl-bash-5.1-p16.drv»
```

changing it to `stdenv.shellPackage` gives

```
$ nix build ".#bison"
error: attribute 'shellPackage' missing

       at /nix/store/czfggskqip31vji4mwgpvgp0474j79yl-source/pkgs/development/tools/parsing/bison/default.nix:27:68:

           26|     # causing infinite recursion during bootstrapping
           27|     (if stdenv.hostPlatform != stdenv.buildPlatform then bash else stdenv.shellPackage)
             |                                                                    ^
           28|   ];
(use '--show-trace' to show detailed location information)
```

Reply:
lets leave it as is for now and then figure it out in a separate pr

Reply:
Sounds good.

Reply:
This is already true by default, but making it explicit sounds ok to me

Reply:
making it explicit prevents a rebuild when `strictDepsByDefault` is true